crates/bevy_hierarchy/src/components/parent.rs

```diff
@@ -8,7 +8,8 @@ use bevy_reflect::Reflect;
 use std::ops::Deref;
 
 /// Holds a reference to the parent entity of this entity.
-/// This component should only be present on entities that actually have a parent entity.
+///
+/// Only present on entities that have a parent.
```

Comment:
Please make a note on how this interacts with `With` and `Without`: `Without` will find root entities. `With` finds any intermediate or leaf entities.